subsys/shell/backends/shell_dummy.c

```diff
@@ -101,9 +101,16 @@ const struct shell_transport_api shell_dummy_transport_api = {
 static int enable_shell_dummy(const struct device *arg)
 {
 	ARG_UNUSED(arg);
+	bool log_backend = CONFIG_SHELL_DUMMY_INIT_LOG_LEVEL > 0;
+	uint32_t level = (CONFIG_SHELL_DUMMY_INIT_LOG_LEVEL > LOG_LEVEL_DBG) ?
+		      CONFIG_LOG_MAX_LEVEL : CONFIG_SHELL_DUMMY_INIT_LOG_LEVEL;
```

Comment:
```suggestion
	uint32_t level = CONFIG_SHELL_DUMMY_INIT_LOG_LEVEL;
```
That check was redundant, KConfig logic will ensure the correct range.

Reply:
This is how they do in Zephyr, it's been copied over. I so this piece of code in multiple shells.

Reply:
DEBUG is level 4, CONFIG_SHELL_DUMMY_INIT_LOG_LEVEL  can go up to 5, and it is not a valid log level.

Reply:
Ok that's strange I'll checkout why they have SHELL_DUMMY_INIT_LOG_LEVEL_DEFAULT ?

Reply:
From the code, it looks like `SHELL_DUMMY_INIT_LOG_LEVEL_DEFAULT` has to be used when want the Log Backend to align with whatever has been set to be the highest log level (ie. `CONFIG_LOG_MAX_LEVEL`)

Reply:
Ah ok, nice! Disregard my original comment then.